### DISCUSSION TITLE
Modified Owncast for RKMPP hardware acceleration (RK3588) successfully, what next?

### DISCUSSION THREAD

**Discussion by okazaki0yumemi1** [owncast/owncast#3494] [Ideas] [answered]
Hello!
I have Orange Pi 5 Plus (RK3588 system-on-a-chip) SBC as a home server with Owncast behind reverse proxy. I am also very new to open-source community, so I have no idea what I should and should not do with code.

The current 0.1.3dev version have no support for RKMPP encoder (hardware RK3588 encoder), but I managed to enable it by adding few lines to `owncast/core/transcoder/codecs.go` file. 

Of course, you need to compile FFMPEG with RKMPP support & make sure you have /dev/rga and /dev/mpp_service. 

I ran a few streams, everything was fine (no dropped frames, no lags and other errors during 2 hour stream). Input stream was:
Nvenc H.264, 1920x1080 @ 60 FPS and 4Mbps bitrate 
output was set to: 1280x720 @ 50FPS and 2Mbps bitrate

I want to help RK3588 users with enabling HW acceleration in Owncast. Should I write instructions for enabling RKMPP in my Github, or I can just fork Owncast project & maintain RKMPP support for as long as I can? What is the best way to help Owncast?

**Comment by gabek** [comment, marked as answer]
Yeah, definitely, it sounds like you're half way there. Thanks for looking into it!

I'm not sure how much you did with `codecs.go`, but you probably saw the pattern of how other codecs are added there.

This is a bit of bad timing, since there's currently a PR up for some changes on this front. So any changes you make will have to be forked off that, but that should be no big deal. You can also look at https://github.com/owncast/owncast/pull/3476 as a model of how to create a new codec model from scratch.

But if you fork the repo. Change to the `gek/ffmpeg-6` branch, create your new branch from that, you can create the new codec in `codec.go` and create the test alongside of its respective `_test.go` file, you can create a PR against the `gek/ffmpeg-6` branch.

I don't have an Orange Pi myself to test, but maybe I can get my hands on one.

**Comment by tsargoth** [comment]
I'd very much like to see such instructions. Thank you

**Reply by gabek** [reply]
It's not about instructions, it's about adding support. And we haven't yet added support. There is a PR to add support, but because it for adding _hardware_ support, and I'm the person who has to support it, I haven't merged it in because I don't have the hardware to support it.

**Reply by okazaki0yumemi1** [reply]
Hi!

I made a fork ( https://github.com/okazaki0yumemi1/owncast-rockchip ) of an old version of Owncast with simple modification. The main problem is to get Rockchip Media Processing Unit to work & to compile ffmpeg - I have a quick guide too (see "What do I need?" chapter). 

I am using Owncast to stream some stuff to my friends, and - so for so good, no complains from them. Direct streaming (no transcoding) with minimum latency is not working though, not sure why.

Also, please tell me if I did something wrong - I am new to FOSS scene and licensing, so please forgive me. 💔

**Reply by gabek** [reply]
Hi there. See https://github.com/owncast/owncast/pull/3869 for the existing PR for Rockchip support.
